odinmain/src/main/kotlin/me/odinmain/features/impl/skyblock/PetKeybinds.kt

```diff
@@ -0,0 +1,70 @@
+import me.odinmain.utils.clock.Clock
+import me.odinmain.utils.name
+import me.odinmain.utils.skyblock.getItemIndexInContainerChestByLore
+import me.odinmain.utils.skyblock.getItemIndexInContainerChestByUUID
+import me.odinmain.utils.skyblock.modMessage
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraftforge.client.event.GuiScreenEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.lwjgl.input.Keyboard
+
+object PetKeybinds : Module(
+    "Pet Keybinds",
+    description = "keybinds for pets (/petkeys)",
+    category = Category.SKYBLOCK
+) {
+    private val unequipKeybind: Keybinding by KeybindSetting("Unequip Keybind", Keyboard.KEY_NONE, "Unequips the current Pet.")
+    private val nextPageKeybind: Keybinding by KeybindSetting("Next Page Keybind", Keyboard.KEY_NONE, "Goes to the next page.")
+    private val previousPageKeybind: Keybinding by KeybindSetting("Previous Page Keybind", Keyboard.KEY_NONE, "Goes to the previous page.")
+    private val delay: Long by NumberSetting("Delay", 0, 0.0, 10000.0, 10.0, description = "The delay between each click .")
+    private val advanced: Boolean by DropdownSetting("Show Settings", false)
+
+    private val pet1: Keybinding by KeybindSetting("Pet 1", Keyboard.KEY_1, "Pet 1 on the list.").withDependency { advanced }
+    private val pet2: Keybinding by KeybindSetting("Pet 2", Keyboard.KEY_2, "Pet 2 on the list.").withDependency { advanced }
+    private val pet3: Keybinding by KeybindSetting("Pet 3", Keyboard.KEY_3, "Pet 3 on the list.").withDependency { advanced }
+    private val pet4: Keybinding by KeybindSetting("Pet 4", Keyboard.KEY_4, "Pet 4 on the list.").withDependency { advanced }
+    private val pet5: Keybinding by KeybindSetting("Pet 5", Keyboard.KEY_5, "Pet 5 on the list.").withDependency { advanced }
+    private val pet6: Keybinding by KeybindSetting("Pet 6", Keyboard.KEY_6, "Pet 6 on the list.").withDependency { advanced }
+    private val pet7: Keybinding by KeybindSetting("Pet 7", Keyboard.KEY_7, "Pet 7 on the list.").withDependency { advanced }
+    private val pet8: Keybinding by KeybindSetting("Pet 8", Keyboard.KEY_8, "Pet 8 on the list.").withDependency { advanced }
+    private val pet9: Keybinding by KeybindSetting("Pet 9", Keyboard.KEY_9, "Pet 9 on the list.").withDependency { advanced }
+
+    private val pets = arrayOf(pet1, pet2, pet3, pet4, pet5, pet6, pet7, pet8, pet9)
+    private val clickCoolDown = Clock(delay)
+
+    val petList: MutableList<String> by ListSetting("List", mutableListOf())
+
+    @SubscribeEvent
+    fun checkKeybinds(event: GuiScreenEvent.KeyboardInputEvent.Pre) {
+        val chest = (event.gui as? GuiChest)?.inventorySlots ?: return
+        if (chest !is ContainerChest) return
+
+        val matchResult = Regex("Pets \\((\\d)/(\\d)\\)").find(chest.name) ?: return
+        val (current, total) = matchResult.destructured
```

Comment:
you sure this works, i would think the first part of the destructured is just the matched string, but if it works its fine

Reply:
worked when i tested it